### PR TITLE
implement auth api

### PR DESCRIPTION
**Pull request by javeme**
Change-Id: Ibd3d4494f5aa3269e9c4f6afd3d11329d8344559

### CODE REVIEW

File: src/main/java/com/baidu/hugegraph/structure/auth/HugePermission.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+        assert code < 256;
+        this.code = (byte) code;
+    }
+
+    public byte code() {
+        return this.code;
+    }
+
+    public String string() {
+        String string = this.name().toLowerCase();
```
Comment (Linary) [inline src/main/java/com/baidu/hugegraph/structure/auth/HugePermission.java:45]: just return

File: src/main/java/com/baidu/hugegraph/structure/auth/HugePermission.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        return string;
+    }
+
+    public boolean match(HugePermission other) {
+        if ((this.code & other.code) != 0) {
```
Comment (Linary) [inline src/main/java/com/baidu/hugegraph/structure/auth/HugePermission.java:50]: return (this.code & other.code) != 0

File: src/main/java/com/baidu/hugegraph/api/auth/AccessAPI.java [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+        super(client, graph);
+    }
+
+    @Override
+    protected String type() {
+        return HugeType.ACCESS.string();
+    }
+
+    public Access create(Access access) {
+        RestResult result = this.client.post(this.path(), access);
```
Comment (zhoney) [inline src/main/java/com/baidu/hugegraph/api/auth/AccessAPI.java:43]: add version check？must >= 0.56？

File: src/test/java/com/baidu/hugegraph/BaseClientTest.java [hunk 4]
```diff
@@ -30,18 +31,24 @@
...
+    protected static final String GRAPH = "hugegraph";
+    protected static final String USERNAME = "admin";
+    protected static final String PASSWORD = "pa";
+    protected static final int TIMEOUT = 10;
 
     private static HugeClient client;
 
     protected static HugeClient open() {
-        return new HugeClient(BASE_URL, GRAPH);
+        client = new HugeClient(BASE_URL, GRAPH, USERNAME, PASSWORD);
```
Comment (zhoney) [inline src/test/java/com/baidu/hugegraph/BaseClientTest.java:43]: just return `new HugeCliennt....`

File: src/main/java/com/baidu/hugegraph/structure/auth/HugePermission.java (hunk 2 above)
Comment (javeme) [reply src/main/java/com/baidu/hugegraph/structure/auth/HugePermission.java:50]: this file is copied from the server code

File: src/test/java/com/baidu/hugegraph/api/auth/AuthApiTest.java [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.baidu.hugegraph.api.auth;
+
+import com.baidu.hugegraph.api.BaseApiTest;
+
+public class AuthApiTest extends BaseApiTest {
```
Comment (zhoney) [inline src/test/java/com/baidu/hugegraph/api/auth/AuthApiTest.java:24]: no use. Still keep it?

File: src/test/java/com/baidu/hugegraph/BaseClientTest.java (hunk 4 above)
Comment (javeme) [reply src/test/java/com/baidu/hugegraph/BaseClientTest.java:43]: to set static field that will be used by other methods like clear()

File: src/test/java/com/baidu/hugegraph/api/auth/AuthApiTest.java (hunk 5 above)
Comment (javeme) [reply src/test/java/com/baidu/hugegraph/api/auth/AuthApiTest.java:24]: keep to do some setup or teardown operations

**Pull request merged**